diff --git a/src/components/widgets/SwitchWidget.vue b/src/components/widgets/SwitchWidget.vue
index 8f8cca6..6e11f0e 100644
--- a/src/components/widgets/SwitchWidget.vue
+++ b/src/components/widgets/SwitchWidget.vue
@@ -1,4 +1,5 @@
 <template>
+<div>
     <li class='widget'
         :class='{dim, loading, "tada": emphasis}'
         :style='style'
@@ -18,6 +19,11 @@
             {{ details }}
         </div>
     </li>
+    <aside v-if='showDetail' @click.prevent>
+        <p>Detail!</p>
+        <button @click.stop='closeDetail'>Close</button>
+    </aside>
+</div>
 </template>
 
 <script>
@@ -42,6 +48,7 @@ export default {
     },
     data () {
         return {
+            showDetail: false,
             style: {
                 backgroundColor: 'maroon'
             }
@@ -52,7 +59,13 @@ export default {
             this.loading = true;
             this.$emit('change', this.value, this.state === 'on' ? 'off' : 'on');
         },
+        closeDetail () {
+            this.showDetail = false;
+            this.$emit('requestFreezeUI', false);
+        },
         longPress () {
+            this.showDetail = true;
+            this.$emit('requestFreezeUI', true);
             console.log(`Long press on ${this.title}.`);
         }
     }
diff --git a/src/components/widgets/Widget.css b/src/components/widgets/Widget.css
index af8867f..573cfa0 100644
--- a/src/components/widgets/Widget.css
+++ b/src/components/widgets/Widget.css
@@ -94,4 +94,20 @@
         height: 10vh;
         box-shadow: 0 -20px rgb(200,200,200);
     }
+}
+ul aside {
+    position: absolute;
+    z-index: 9;
+    width: 78vw;
+    height: 66vh;
+    left: 7vw;
+    top: 5vh;
+    padding: 2vh 3vw;
+    background-color: #1a1a1a;
+    border: 2px solid #333;
+    border-radius: 8px;
+}
+ul aside,
+ul aside button {
+    color: #dfdfdf;
 }
\ No newline at end of file
